Source/Evolve/WarpXEvolve.cpp

```diff
@@ -494,7 +504,29 @@ WarpX::OneStep_nosub (Real cur_time)
     ExecutePythonCallback("particlescraper");
     ExecutePythonCallback("beforedeposition");
 
-    PushParticlesandDeposit(cur_time);
+    // push particles (half position and full momentum)
+    PushParticlesandDeposit(
+        a_cur_time,
+        /*skip_current=*/true,
+        PositionPushType::FirstHalf,
+        MomentumPushType::Full
+    );
+
+    // communicate particle data
+    mypc->Redistribute();
+
+    // perform particle collisions
+    ExecutePythonCallback("beforecollisions");
+    mypc->doCollisions(a_step, a_cur_time, a_dt);
+    ExecutePythonCallback("aftercollisions");
+
+    // push particles (half position)
+    PushParticlesandDeposit(
+        a_cur_time,
+        /*skip_current=*/false,
+        PositionPushType::SecondHalf,
+        MomentumPushType::None
+    );
```

Comment:
This needs to take into account the value of the user input `m_collisions_split_position_push`.